django/db/models/base.py

```diff
@@ -873,7 +898,11 @@ def save_base(
         with context_manager:
             parent_inserted = False
             if not raw:
-                parent_inserted = self._save_parents(cls, using, update_fields)
+                # validate force insert only if parents are inserted
+                force_insert = self._validate_force_insert(force_insert)
```

Comment:
I think that's a compromise worth doing, that limits the possible extra cost of  ` _validate_force_insert` to MTI inserts 🎉